packages/components/pie-toggle-switch/src/index.ts

```diff
@@ -9,13 +14,51 @@ export {
 
 const componentSelector = 'pie-toggle-switch';
 
-export class PieToggleSwitch extends LitElement implements ToggleSwitchProps {
-    render () {
-        return html`<h1>Hello world!</h1>`;
-    }
+/**
+ * @event {CustomEvent} pie-toggle-switch-changed - when the toggle switch checked state is changed.
+ */
+
+export class PieToggleSwitch extends RtlMixin(LitElement) implements ToggleSwitchProps {
+    @property({ type: Boolean, reflect: true })
+    public isChecked = false;
+
+    @property({ type: Boolean, reflect: true })
+    public isDisabled = false;
 
-    // Renders a `CSSResult` generated from SCSS by Vite
     static styles = unsafeCSS(styles);
+
+    onToggleChange (event: Event) {
+        const target = event?.target as HTMLInputElement;
+        this.isChecked = target.checked;
+        this.dispatchEvent(new CustomEvent(EVENT_TOGGLE_SWITCH_CHANGED, { detail: this.isChecked }));
+    }
+
+    render () {
+        const {
+            isChecked,
+            isDisabled,
+        } = this;
+
+        return html`
+            <label
+                data-test-id="toggle-switch-component"
+                class="c-toggle-switch"
+                ?isChecked="${isChecked}"
+                ?isDisabled=${isDisabled}>
+                <input
+                    role="switch"
+                    type="checkbox"
+                    class="c-toggle-switch-input"
+                    .checked="${isChecked}"
+                    .disabled="${isDisabled}"
+                    @change="${this.onToggleChange}">
+
+                <div class="c-toggle-switch-control">
+                    ${isChecked ? html`<icon-check />` : nothing}
```

Comment:
This should be: (we were doing it wrong previously on other components)

```suggestion
                    ${isChecked ? html`<icon-check></icon-check>` : nothing}
```